third_party/freenove/smart_car/controller.py

```diff
@@ -65,24 +65,48 @@ def __init__(self):
         GPIO.setup(self.IR02, GPIO.IN)
         GPIO.setup(self.IR03, GPIO.IN)
 
-    def read(self, position):
+    def read(self):
         """
         Parameters
         ----------
         position: 1 to 3, There's number on the board.
         -------
         """
-        if position == 1:
-            if GPIO.input(self.IR01):
-                print("Left has been detected on bright")
-        elif position == 2:
-            if GPIO.input(self.IR02):
-                print("Middle has been detected on bright")
-        elif position == 3:
-            if GPIO.input(self.IR03):
-                print("Right has been detected on bright")
+        # if position == 1:
+        #     if GPIO.input(self.IR01):
+        #         print("Left has been detected on bright")
+        # elif position == 2:
+        #     if GPIO.input(self.IR02):
+        #         print("Middle has been detected on bright")
+        # elif position == 3:
+        #     if GPIO.input(self.IR03):
+        #         print("Right has been detected on bright")
+        # else:
+        #     print("Nothing has been detected.")
+        if GPIO.input(self.IR01) and GPIO.input(self.IR02) and GPIO.input(self.IR03):
+            # print('LMR')
+            return [0, 1, 2]
+        elif GPIO.input(self.IR01) and GPIO.input(self.IR02):
+            # print('LM')
+            return [0, 1]
+        elif GPIO.input(self.IR01) and GPIO.input(self.IR03):
+            # print('LR')
+            return [0, 2]
+        elif GPIO.input(self.IR02) and GPIO.input(self.IR03):
+            # print('MR')
+            return [1, 2]
+        elif GPIO.input(self.IR01):
+            # print('L')
+            return [0]
+        elif GPIO.input(self.IR02):
+            # print('M')
+            return [1]
+        elif GPIO.input(self.IR03):
```

Comment:
Even thought this works, its not a good practice to code as such. One approach is as follows:



```GPIO_input = [GPIO.input(self.IR01), GPIO.input(self.IR02), GPIO.input(self.IR03)]
GPIO_state = list[]
for input in GPIO_input:
      if input:
            GPIO_state.append(GPIO_input.index[input])```